medicines/doc-index-updater/src/create_manager/models.rs

```diff
@@ -1,43 +1,138 @@
+        let mut metadata: HashMap<String, String> = HashMap::new();
 
-    let active_substances = to_json(document.active_substances.clone());
-    metadata.insert("substance_name".to_string(), active_substances);
+        metadata.insert("file_name".to_string(), self.file_name.clone());
+        metadata.insert("doc_type".to_string(), self.doc_type.clone().to_string());
+        metadata.insert("title".to_string(), self.title.clone());
+        metadata.insert(
+            "product_name".to_string(),
+            to_json(self.product_names.clone()),
+        );
+        metadata.insert(
+            "substance_name".to_string(),
+            to_json(self.active_substances.clone()),
+        );
+        metadata.insert("facets".to_string(), to_json(self.facets()));
+        if let Some(keywords) = self.keywords.clone() {
+            metadata.insert("keywords".to_string(), keywords.join(" "));
+        }
+        metadata.insert("pl_number".to_string(), self.pl_number.clone());
+        metadata.insert("author".to_string(), self.author.clone());
 
-    let facets = to_json(create_facets_by_active_substance(
-        &product_names_csv,
-        document.active_substances.clone(),
-    ));
-    metadata.insert("facets".to_string(), facets);
+        metadata
+    }
+}
 
-    let author = sanitize(&document.author);
-    metadata.insert("author".to_string(), author);
+#[derive(Debug, Serialize)]
+pub struct IndexEntry {
+    content: String,
+    rev_label: String,
+    metadata_storage_path: String,
+    metadata_content_type: String,
+    product_name: String,
+    metadata_language: String,
+    created: String,
+    release_state: String,
+    keywords: String,
+    title: String,
+    pl_number: Vec<String>,
+    file_name: String,
+    metadata_storage_content_type: String,
+    metadata_storage_size: usize,
+    metadata_storage_last_modified: String,
+    metadata_storage_content_md5: String,
+    metadata_storage_name: String,
+    doc_type: String,
+    suggestions: Vec<String>,
+    substance_name: Vec<String>,
+    facets: Vec<String>,
+    is_deleted: bool,
+}
 
-    if let Some(keywords) = &document.keywords {
-        metadata.insert("keywords".to_string(), keywords.join(" "));
+impl IndexEntry {
+    pub fn for_blob(
+        metadata_storage_name: String,
+        blob: BlobMetadata,
+        file_size: usize,
+        storage_path: String,
+    ) -> Self {
+        Self {
+            content: "Content not yet available".to_owned(),
+            rev_label: "1".to_owned(),
+            product_name: blob.product_names.join(", "),
+            created: Utc::now().to_rfc3339_opts(SecondsFormat::Secs, true),
+            release_state: "Y".to_owned(),
+            keywords: blob.keywords.to_owned().unwrap_or_default().join(", "),
+            title: blob.title.to_owned(),
+            pl_number: vec![blob.pl_number.to_owned()],
+            file_name: blob.file_name.to_owned(),
+            doc_type: blob.doc_type.to_string(),
```

Comment:
Why the mix of `.to_owned()` and `.to_string()` here? Aren't they the same thing?

Reply:
Absolutely not. `doc_type` is a `DocumentType`, so I need to convert it to a `String`. The others are `String` properties of the `blob`, and I need to let the new `IndexEntry` own a copy of them, so I say `to_owned()`.